### ISSUE TITLE
ci: sandbox seems not work for internal stage.

### ISSUE THREAD

**Issue by youngsofun** [databendlabs/databend#12980]
**Summary**

the path of stage object meta in the meta server includes tenant.
but the location in "internal storage" does not.

```
    pub fn stage_prefix(&self) -> String {
        match self.stage_type {
            StageType::LegacyInternal => format!("/stage/{}/", self.stage_name),
            StageType::External => {
                unreachable!("stage_prefix should never be called on external stage, must be a bug")
            }
            StageType::Internal => format!("/stage/internal/{}/", self.stage_name),
            StageType::User => format!("/stage/user/{}/", self.stage_name),
        }
```

is this by design?

so sqllogict tests with sandbox (each sandbox with a diff tenant for isolation) need to `remove @stage` each time to make sure it is not empty,
especially for we run a same tests for 3 handlers.
e.g. https://github.com/datafuselabs/databend/pull/12968#issuecomment-1731453003

or maybe for internal stage we should remove file at the same time of drop stage?

**Comment by youngsofun** [comment]
cc @Xuanwo @flaneur2020 @BohuTANG

**Comment by BohuTANG** [comment]
I think @Xuanwo will know that.

For tenant:
```
create stage s1;
```
The stage `s1` prefix will be?

**Comment by youngsofun** [comment]
`"format("/stage/internal/{}/", self.stage_name)`

**Comment by BohuTANG** [comment]
In cloud, the root prefix will be the tenant, so the final prefix will be:
`/tenant/stage/internal/stage_name`

**Comment by youngsofun** [comment]
open another issue https://github.com/datafuselabs/databend/issues/12986

**Comment by Xuanwo** [comment]
> the path of stage object meta in the meta server includes tenant.
> but the location in "internal storage" does not.

It's not by design. We should fix them.

**Comment by Xuanwo** [comment]
> or maybe for internal stage we should remove file at the same time of drop stage?

We don't remove files in external stage. But it makes sense to remove files in internal stage since users can't operate them after stage got dropped.

**Comment by youngsofun** [comment]
> It's not by design. We should fix them.

for compatibility, we can add an option FOR_CURRENT_TENANT  when create internal stage.
FOR_CURRENT_TENANT=true by default
set FOR_CURRENT_TENANT=false on cloud
make sure FOR_CURRENT_TENANT=false for eixsting internal stages.

**Comment by BohuTANG** [comment]
> > It's not by design. We should fix them.
> 
> for compatibility, we can add an option FOR_CURRENT_TENANT when create internal stage. FOR_CURRENT_TENANT=true by default set FOR_CURRENT_TENANT=false on cloud make sure FOR_CURRENT_TENANT=false for eixsting internal stages.

I think cloud doesn't care this issue, we already put tenant id to each stage prefix. 
cc @everpcpc for confirm

**Comment by youngsofun** [comment]
> > > It's not by design. We should fix them.
> 
> > 
> 
> > for compatibility, we can add an option FOR_CURRENT_TENANT when create internal stage. FOR_CURRENT_TENANT=true by default set FOR_CURRENT_TENANT=false on cloud make sure FOR_CURRENT_TENANT=false for eixsting internal stages.
> 
> 
> 
> I think cloud doesn't care this issue, we already put tenant id to each stage prefix. Why we need 'FOR_CURRENT_TENANT' here?
> 
> cc @everpcpc for confirm

not have to， but may contain 2 tenant in the path

**Comment by everpcpc** [comment]
> > > It's not by design. We should fix them.
> > 
> > 
> > for compatibility, we can add an option FOR_CURRENT_TENANT when create internal stage. FOR_CURRENT_TENANT=true by default set FOR_CURRENT_TENANT=false on cloud make sure FOR_CURRENT_TENANT=false for eixsting internal stages.
> 
> I think cloud doesn't care this issue, we already put tenant id to each stage prefix. Why we need 'FOR_CURRENT_TENANT' here? cc @everpcpc for confirm

Yes, we have different s3 prefix for every tenant at storage level.

**Comment by everpcpc** [comment]
> but may contain 2 tenant in the path

That's not a good idea.

**Comment by youngsofun** [comment]
![image](https://github.com/datafuselabs/databend/assets/5782159/bc8bd44c-ae4f-492f-bccb-2ef051ab2a1a)

we now store table and internal stage data in the configured storage.
both do not include tenant in the path.
but table data is indexed by table id,   which avoids this issue.

**Comment by everpcpc** [comment]
> but table data is indexed by table id, which avoids this issue.
Yes, that's a lucky case for test, but we should not count on that.
Since we are using sandbox tenant in the same instance without modifying the config for test, I suggest we should store data for sandboxed tenant in a separate prefixed directory in storage.

**Comment by youngsofun** [comment]
> > It's not by design. We should fix them.
> 
> for compatibility, we can add an option FOR_CURRENT_TENANT when create internal stage. FOR_CURRENT_TENANT=true by default set FOR_CURRENT_TENANT=false on cloud make sure FOR_CURRENT_TENANT=false for eixsting internal stages.

@everpcpc yes,
so we need make this change?

**Comment by everpcpc** [comment]
> @everpcpc yes, so we need make this change?

I don't think so. Sandbox tenant is a feature only for test, we could modify it easily. Stage creation should not care about that.
